File: backend-ulysses/src/main/java/com/luisa13/backendulysses/service/UserService.java
package com.luisa13.backendulysses.service;

import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import com.luisa13.backendulysses.model.User;
import com.luisa13.backendulysses.repository.UserRepository;

@Component
public class UserService implements IUserService{

	@Autowired
	private UserRepository userRepo;

	@Override
	public User addUser(User user) {
		return this.userRepo.save(user);
	}

	@Override
	public void deleteUser(User user) {
		this.userRepo.delete(user);
	}

	@Override
	public User findUserById(Long id) {
		return this.userRepo.findById(id).get();
	}

	@Override
	public User updateUser(User user) {
		User oldUser = this.userRepo.findById(user.getId())
				.orElseThrow();
		userRepo.save(user);
		return user;
	}

	@Override
	public List<User> getAllUsers() {
		return this.userRepo.findAll();
	}

	@Override
	public void deleteUserById(Long id) {
		this.userRepo.deleteById(id);
		
	}

	@Override
	public User findUserByName(String name) {
		return this.userRepo.findByName(name).get();
	}
	
	

}

File: client-ulysses/src/domain/service/UserService.ts
import User from '../entity/User';
import IUserRepository from '../repository/IUserRepository';


export interface IUserService{
    getUsers():Promise<User[]>;
    updateUser(user:object): Promise<Response>
}

export default class UserService implements IUserService{

    userRepo: IUserRepository;

    constructor(ur: IUserRepository){
        this.userRepo = ur;
    }

    async getUsers():Promise<User[]>{
        return this.userRepo.getUsers();
    }

    async updateUser(user:object): Promise<Response>{
        return this.userRepo.updateUser(user);
    }
}